test/camunda-cloud/error-reference.spec.js

```diff
@@ -89,6 +131,34 @@ const invalid = [
         requiredProperty: 'errorCode'
       }
     }
+  },
+  {
+    name: 'error throw event (error code as expression Camunda 8.1)',
+    config: { version: '8.1' },
+    moddleElement: createModdle(createDefinitions(`
+      <bpmn:process id="Process_1">
+        <bpmn:endEvent id="Event">
+          <bpmn:errorEventDefinition id="ErrorEventDefinition" errorRef="Error" />
+        </bpmn:endEvent>
+      </bpmn:process>
+      <bpmn:error id="Error" name="Error" errorCode="=myCode" />
+    `)),
+    report: {
+      id: 'Event',
+      message: 'Property value of <=myCode> only allowed by Camunda Platform 8.2 or newer',
```

Comment:
I'm wondering if we can be a little bit more descriptive here:

`=myCode` is an expression, and as such it is not supported. It is not clear from the message that the problem is the fact that `errorCode` is an expression statement.

A little bit more helpful could be something along these lines:

```
Expression statement <=myCode> only supported by Camunda Platform 8.2 or newer
```

Reply:
The human-readable messages are provided in `camunda/linting` project: https://github.com/camunda/linting/blob/main/lib/utils/error-messages.js
I will provide a message similar to your suggestion there :)

Reply:
Of course it's open to discussion whether it makes sense to have a purely technical plugin like this one and a separate project which overwrites the error messages.

Reply:
I'm not sold on "human readable plugged in somewhere else". The errors produced by this plug in should absolutely be human readable, so that they can be fixed when being reported via command line. If the error message misses important context, I don't know how to fix it.

What [`@camunda/linting`](https://github.com/camunda/linting/blob/main/lib/utils/error-messages.js) adds on top is errors that are contextual, during modeling.

Reply:
How about if we have a specialized rule `no-expression`:

> Property <errorCode> can be an expression in Camunda Platform 8.2 or newer.

Reply:
:+1:

Reply:
> How about if we have a specialized rule no-expression:

I'd assume it would provide relevant context, also via CLI? The benefit of `error-reference` is that the context is clear.

Reply:
It would be a general rule like `element`: element type is not supported; respectively: property as FEEL expression not supported.

Reply:
Still, in the CLI you get element's ID, property name etc.

Reply:
The error messages should be human-readable in the CLI. `@camunda/linting` adjusts the error messages to refer to groups and entries in the properties panel.